src/main/java/org/grouplens/grapht/reflect/internal/ClassInstantiator.java

```diff
@@ -74,7 +78,10 @@ public Class getType() {
     @Override
     public Object instantiate() throws ConstructionException {
         // find constructor and build up necessary constructor arguments
+
+        LogContext mdcContext = LogContext.create();
```

Comment:
We need a `try` to start here, with its `finally` closing the context; in this case, if the code fails before the `try`, then the context will not be finished.